tests/test_utils.py

```diff
@@ -66,6 +66,32 @@ def test_file(self, tmpdir, mode, warnings_num, recwarn):
             assert super_len(fd) == 4
         assert len(recwarn) == warnings_num
 
+    def test_super_len_with__len__(self):
+        foo = [1,2,3,4]
+        len_foo = super_len(foo)
+        assert len_foo == 4
+
+    def test_super_len_with_no__len__(self):
+        foo = BytesIO(b'1234')
+        assert getattr(foo, '__len__', None) is None
+        setattr(foo, 'len', 4)
+        length = foo.len
+        assert length == 4
+        assert super_len(foo) == length
+
+    def test_super_len_with_tell(self):
+        foo = StringIO.StringIO('12345')
+        assert super_len(foo) == 5
+        foo.read(2)
+        assert super_len(foo) == 3
+
+    def test_super_len_with_fileno(self):
+        f = open(__file__, 'rb')
+        length = super_len(f)
+        file_data = f.read()
+        f.close()
+        assert length == len(file_data)
+
 
```

Comment:
I was going to add a test for a file-like object that didn't have any of the attributes (`__len__`, `len`, `fileno`, or `tell`/`seek`) to make sure it defaults to 0. I can't delete `seek` or `tell` from any of the io classes, or default file object though. Any suggestions on how to simulate this?